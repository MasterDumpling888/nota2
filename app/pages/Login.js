import React, { useState } from 'react';
import { View, Text, TextInput, StyleSheet, TouchableOpacity } from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';
import PageBox from '../components/PageBox';
import { signInWithEmailAndPassword } from 'firebase/auth';
import { auth } from '../firebaseConfig';
import { getFontSize } from '../responsiveFont';

function Login({ navigation }) {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');

  const handleLogin = async () => {
    try {
      const userCredential = await signInWithEmailAndPassword(auth, email, password);
      const user = userCredential.user;
      // Navigate to the homepage
      navigation.navigate('Home');
    } catch (error) {
      console.error(error);
      // Handle login error
    }
  };

  return (
    <SafeAreaView style={{ flex: 1 }}>
      <PageBox title="Login" onClose={() => navigation.goBack()}>
        <View style={styles.inputContainer}>
          <Text style={styles.inputText}>Email</Text>
          <TextInput
            style={styles.input}
            value={email}
            onChangeText={setEmail}
            keyboardType="email-address"
            autoCapitalize="none"
            placeholder='Type in your email here'
          />
        </View>
        <View style={styles.inputContainer}>
          <Text style={styles.inputText}>Password</Text>
          <TextInput
            style={styles.input}
            value={password}
            onChangeText={setPassword}
            secureTextEntry
          />
        </View>
        <TouchableOpacity style={styles.loginButton} onPress={handleLogin}>
          <Text style={styles.loginText}>Login</Text>
        </TouchableOpacity>
      </PageBox>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  inputContainer: {
    marginBottom: 16,
  },
  input: {
    borderWidth: 1,
    borderColor: '#32FB0A',
    borderRadius: 5,
    padding: 8,
    marginTop: 8,
    backgroundColor: '#CFFFC5'
  },
  inputText: {
    fontSize: getFontSize(14),
  },
  loginButton: {
    backgroundColor: '#32FB0A',
    justifyContent: 'center',
    alignItems: 'center',
    padding: 16,
    borderRadius: 5
  },
  loginText: {
    fontSize: getFontSize(16),
    fontWeight: 'bold'
  }
});

export default Login;
