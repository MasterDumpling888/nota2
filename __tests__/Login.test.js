import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import Login from '../app/pages/Login';

test('renders Email input', () => {
    const { getByText } = render(<Login navigation={global.mockNavigation} />);
    expect(getByText('Email')).toBeTruthy();
});

test('Login snapshot test', () => {
    const { toJSON } = render(<Login navigation={global.mockNavigation} />);
    expect(toJSON()).toMatchSnapshot();
});