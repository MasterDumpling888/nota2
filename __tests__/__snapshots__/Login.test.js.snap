// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login snapshot test 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#181818",
        "flex": 1,
        "padding": 24,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "opacity": 1,
          }
        }
      >
        <Icon
          color="#33FD0A"
          name="chevron-back-outline"
          size={24}
          style={
            {
              "marginBottom": 15,
              "padding": 0,
            }
          }
        />
        <Text
          style={
            {
              "color": "#33FD0A",
              "fontFamily": "Raleway-SemiBold",
              "fontSize": 17,
              "marginBottom": 16,
            }
          }
        >
          Login
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "marginBottom": 16,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontFamily": "Raleway-Regular",
            "fontSize": 13,
            "textAlign": "left",
          }
        }
      >
        Email
      </Text>
      <TextInput
        autoCapitalize="none"
        keyboardType="email-address"
        onChangeText={[Function]}
        style={
          {
            "borderColor": "#33FD0A",
            "borderRadius": 5,
            "borderWidth": 1,
            "color": "white",
            "fontFamily": "Raleway-SemiBold",
            "fontSize": 13,
            "marginTop": 8,
            "padding": 8,
            "textAlign": "left",
          }
        }
        value=""
      />
    </View>
    <View
      style={
        {
          "marginBottom": 16,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontFamily": "Raleway-Regular",
            "fontSize": 13,
            "textAlign": "left",
          }
        }
      >
        Password
      </Text>
      <TextInput
        onChangeText={[Function]}
        secureTextEntry={true}
        style={
          {
            "borderColor": "#33FD0A",
            "borderRadius": 5,
            "borderWidth": 1,
            "color": "white",
            "fontFamily": "Raleway-SemiBold",
            "fontSize": 13,
            "marginTop": 8,
            "padding": 8,
            "textAlign": "left",
          }
        }
        value=""
      />
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#33FD0A",
          "borderRadius": 5,
          "justifyContent": "center",
          "opacity": 1,
          "padding": 16,
        }
      }
    >
      <Text
        style={
          {
            "fontFamily": "Raleway-SemiBold",
            "fontSize": 15,
          }
        }
      >
        Login
      </Text>
    </View>
  </View>
</RNCSafeAreaView>
`;
